evap/evaluation/management/commands/anonymize.py

```diff
@@ -111,22 +129,100 @@ def replace_usernames_emails(random_usernames):
                 new_domain = Command.new_institution_domain if is_institution_domain else Command.new_external_domain
                 user.email = user.username + '@' + new_domain
 
-            user.save()
+            user.login_key = key
+            user.login_key_valid_until = valid_until
 
-    @staticmethod
-    def shuffle_courses():
-        # do this per semester to avoid problems e.g. with archived semesters
+            user.save()            
+
+
+    # Comprehensively anonymizes courses.
+    def anonymize_courses(self):
+        all_degrees = Degree.objects.all()
+        all_course_types = CourseType.objects.all()
+
+        # Handling courses of each semester separately in order to prevent
+        # conflicts with archived courses.
         for semester in Semester.objects.all():
-            shuffled_courses = list(semester.courses.all())
-            random.shuffle(shuffled_courses)
-
-            for i, course in enumerate(semester.courses.all()):
-                course.degrees.set(shuffled_courses[i].degrees.all())
-                course.semester = shuffled_courses[i].semester
-                course.name_de = shuffled_courses[i].name_de + " "  # add a space to avoid name collisions
-                course.name_en = shuffled_courses[i].name_en + " "
+            courses = semester.courses.all()
+            public_courses = [course for course in courses if not course.is_private]
+            private_courses = [course for course in courses if course.is_private]
+
+            num_public, num_private = len(public_courses), len(private_courses)
+            self.stdout.write(f"Anonymizing courses of semester {semester} ("
+                + ( f"{num_public} public" if num_private == 0 else 
+                    f"{num_private} private" if num_public == 0 else
+                    f"{num_public} public, {num_private} private, {len(courses)} in total" )
```

Comment:
lets keep it simple and always print out the three numbers. it's an admin tool, no need to impress anyone ;)